From 6c3728cc93a767b6b97d0daa21985fc496f7af83 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Fri, 23 Sep 2022 16:43:31 +0300
Subject: [PATCH] [SW-104919] gaudi2: security: fix for ppboot hang

This patch fixes hang issue seen in ppboot when real
security is enabled. The hang is observed to be in
arcboot_rom's memcmp() function called from rsa_verify().
The root cause of this issue is not found yet. But it is
observed if https://gerrit.habana-labs.com/#/c/254030/ is
reverted issue is not reproduced. Alternatively doing the
modification done in this patch also solves the issue
while it keeps the changes done in that patch. So this is
pushed to unblock enablement of real security.

TODO: Need to root cause the issue to avoid similar issue in future
(https://jira.habana-labs.com/browse/SW-105874 is raised).

Change-Id: I395920fcb5a11daf9505df1ef92fb8a8abad1cad
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/arch/gaudi2/arc/common.c b/arch/gaudi2/arc/common.c
index cf27ce2..9f220a9 100644
--- a/arch/gaudi2/arc/common.c
+++ b/arch/gaudi2/arc/common.c
@@ -335,13 +335,16 @@
 	return 0;
 }
 
+/* TODO: This does not need to be global. Make it local when
+ * https://jira.habana-labs.com/browse/SW-105874 is concluded.
+ */
+static uint32_t magic;
 static int ppboot_bringup_preboot(u8 image_ptr)
 {
 	u32 image_size, payload_size, copy_size;
 	u32 flash_offset;
 	u32 num_cycles;
 	int ret;
-	uint32_t magic = 0;
 	uint8_t hash[SHA384_SUM_LEN] = { 0 };
 
 	flash_offset = ppboot_flash_image_offset(image_ptr);
